### PR TITLE
Added support for GZIP compressed pages

### PR DESCRIPTION
**Pull request by halilozercan**
If Content-Encoding is gzip, ```bytes[] data``` should not be directly used. 

Added a new method called decompressGZIP and used it in asBytes if response contained gzip Content-Encoding.

### CODE REVIEW

File: library/src/main/java/com/afollestad/bridge/Response.java [hunk 1]
```diff
@@ -112,7 +123,19 @@ public boolean isSuccess() {
 
     @Nullable
     public byte[] asBytes() {
-        return data;
+        String encoding = contentEncoding();
+        if(encoding!=null && encoding.contains("gzip")){
```
Comment (afollestad) [inline library/src/main/java/com/afollestad/bridge/Response.java]: Can you format code in this file? The if statement and try/catch statement aren't using standard formatting.

File: library/src/main/java/com/afollestad/bridge/Response.java [hunk 2]
```diff
@@ -276,4 +300,22 @@ public String toString() {
...
+        ByteArrayInputStream is = new ByteArrayInputStream(compressed);
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        GZIPInputStream gis = new GZIPInputStream(is, BUFFER_SIZE_GZIP);
+
+        byte[] data = new byte[BUFFER_SIZE_GZIP];
+        int bytesRead;
+        while ((bytesRead = gis.read(data)) != -1) {
+            os.write(data, 0, bytesRead);
+        }
+        gis.close();
```
Comment (afollestad) [inline library/src/main/java/com/afollestad/bridge/Response.java]: Everything except the `close()` statements should go in a `try` block, while the `close()` statements should go in a `finally` block. That way the streams are closed even if an exception is thrown to the method.

Comment (halilozercan) [inline library/src/main/java/com/afollestad/bridge/Response.java]: alright, I'll be making necessary changes.

**Comment by afollestad** [comment]
Thanks!

**Pull request merged**